### PR TITLE
pythranrc check

### PR DESCRIPTION
**Pull request by ashwinvis**
Relates to issue #1117.

With this implementation we get output like:

```bash
❯❯❯ pythran-config -v                                                                                                                            (pythran) enh/pythranrc-sanity-check
WARNING  pythranrc section [pythran] is invalid!
WARNING  pythranrc section [pythran] options: {'cc', 'blas', 'ignoreflags', 'cxx'} are invalid!
WARNING  pythranrc section [dummy] is invalid!
c++ -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/home/avmo/src/pythran/pythran -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/scratch/avmo/opt/pythran/include -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m -L/usr/lib64 -lcblas -lblas -L/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu -lpthread -ldl -lutil -lpython3.7

❯❯❯ pythran-config -vv                                                                                                                           (pythran) enh/pythranrc-sanity-check
INFO     sys file exists: True 
	/home/avmo/src/pythran/pythran/pythran.cfg
INFO     platform file exists: True 
	/home/avmo/src/pythran/pythran/pythran-linux.cfg
INFO     user file exists: True 
	/home/avmo/.pythranrc
WARNING  pythranrc section [pythran] is invalid!
WARNING  pythranrc section [pythran] options: {'cc', 'blas', 'cxx', 'ignoreflags'} are invalid!
INFO     pythranrc section [typing] is valid
INFO     pythranrc section [compiler] is valid
WARNING  pythranrc section [dummy] is invalid!
     CXX = c++ 
  CFLAGS = -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/home/avmo/src/pythran/pythran -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/scratch/avmo/opt/pythran/include -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m 
 LDFLAGS = -L/usr/lib64 -lcblas -lblas -L/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu -lpthread -ldl -lutil -lpython3.7
```

Also for pythran command:

```sh
 ❯❯❯ pythran -v euler01.py                                                                                                                        (pythran) enh/pythranrc-sanity-check
INFO     sys file exists: True 
	/home/avmo/src/pythran/pythran/pythran.cfg
INFO     platform file exists: True 
	/home/avmo/src/pythran/pythran/pythran-linux.cfg
INFO     user file exists: True 
	/home/avmo/.pythranrc
WARNING  pythranrc section [pythran] is invalid!
WARNING  pythranrc section [pythran] options: {'cc', 'ignoreflags', 'blas', 'cxx'} are invalid!
INFO     pythranrc section [typing] is valid
INFO     pythranrc section [compiler] is valid
WARNING  pythranrc section [dummy] is invalid!
CRITICAL I am in trouble. Your input file does not seem to match Pythran's constraints...
E: identifier xrange unknown, either because it is an unsupported intrinsic, the input code is faulty, or... pythran is buggy. (line 11)
```

### CODE REVIEW

**Comment by ashwinvis** [comment]
Improved messages for clarity:

```sh
~/s/p/p/t/euler ❯❯❯ pythran-config -v                                                                                                                     (pythran)  ✘ 130 enh/pythranrc-sanity-check
WARNING  pythranrc section [pythran] is valid but options are incorrect!
WARNING  pythranrc section [pythran] options: {'ignoreflags', 'blas', 'cc', 'cxx'} are incorrect!
WARNING  pythranrc section [dummy] is invalid!
c++ -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/home/avmo/src/pythran/pythran -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/scratch/avmo/opt/pythran/include -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m -L/usr/lib64 -lcblas -lblas -L/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu -lpthread -ldl -lutil -lpython3.7
~/s/p/p/t/euler ❯❯❯ pythran-config -vv                                                                                                                           (pythran) enh/pythranrc-sanity-check
INFO     sys file exists: True 
	/home/avmo/src/pythran/pythran/pythran.cfg
INFO     platform file exists: True 
	/home/avmo/src/pythran/pythran/pythran-linux.cfg
INFO     user file exists: True 
	/home/avmo/.pythranrc
WARNING  pythranrc section [pythran] is valid but options are incorrect!
WARNING  pythranrc section [pythran] options: {'cc', 'cxx', 'ignoreflags', 'blas'} are incorrect!
INFO     pythranrc section [typing] is valid and options are correct
INFO     pythranrc section [compiler] is valid and options are correct
WARNING  pythranrc section [dummy] is invalid!
     CXX = c++ 
  CFLAGS = -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/home/avmo/src/pythran/pythran -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/scratch/avmo/opt/pythran/include -I/scratch/avmo/opt/pythran/lib/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m 
 LDFLAGS = -L/usr/lib64 -lcblas -lblas -L/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu -lpthread -ldl -lutil -lpython3.7
```

File: pythran/config.py [hunk 1]
```diff
@@ -21,20 +26,67 @@ def init_cfg(sys_file, platform_file, user_file):
...
+
+    # Use configuration from sys and platform as "reference"
+    cfgp_ref = ConfigParser()
+    cfgp_ref.read([paths["sys"], paths["platform"]])
+
+    # Check if pythran configuration files exists
+    for loc, path in paths.items():
+        exists = os.path.exists(path)
+
+        msg = "{} file {} \n\t ↳ {}".format(
```
Comment (serge-sans-paille) [inline pythran/config.py]: ↳ is a non-ascii character, and it makes the python2 build crash. Can you use another formating.
I would replace

```
INFO     sys file exists 
	 ↳ /home/avmo/src/pythran/pythran/pythran.cfg
```

by

```
INFO     sys file exists:  /home/avmo/src/pythran/pythran/pythran.cfg
```

sorry for the eye candies ;-)

File: pythran/config.py [hunk 2]
```diff
@@ -21,20 +26,67 @@ def init_cfg(sys_file, platform_file, user_file):
...
-    for obsolete_section in ('user', 'sys'):
-        if cfgp.has_section(obsolete_section):
-            logger.warn("Your pythranrc has an obsolete `%s' section",
-                        obsolete_section)
-
     return cfgp
 
 
+def lint_cfg(cfgp, **paths):
+    if len(paths) == 0:
```
Comment (serge-sans-paille) [inline pythran/config.py]: ``if not paths:`` looks more pythonic

File: pythran/config.py [hunk 3]
```diff
@@ -1,8 +1,9 @@
+from __future__ import unicode_literals
 try:
     # python3 vs. python2
-    import configparser
+    from configparser import ConfigParser
 except ImportError:
-    import ConfigParser as configparser
+    from ConfigParser import SafeConfigParser as ConfigParser
```
Comment (serge-sans-paille) [inline pythran/config.py]: why this change?

Comment (ashwinvis) [reply pythran/config.py]: To avoid:
```python
>>> cfg = SafeConfigParser().read(["~/.pythranrc"])
__main__:1: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
```

File: pythran/config.py (hunk 1 above)
Comment (ashwinvis) [reply pythran/config.py]: I had hoped `from __future__ import unicode_literals` this would manage it... :disappointed: . Let's do it after 2020 then :wink:

File: pythran/config.py (hunk 3 above)
Comment (serge-sans-paille) [reply pythran/config.py]: cool, thanks!

File: pythran/config.py [hunk 4]
```diff
@@ -170,9 +228,22 @@ def run():
...
 
+    if args.verbose >= 1:
+        logger.setLevel(max(logging.DEBUG, logging.ERROR - args.verbose * 10))
+        lint_cfg(cfg)
+        args.compiler = True
+        args.cflags = True
+        args.libs = True
+
+    if args.verbose >= 2:
+        output.append("CXX =".rjust(10))
```
Comment (serge-sans-paille) [inline pythran/config.py]: `output` does to stdout. that's not waht you want, if I'm not mistaken; Why not just log this info?

**Comment by serge-sans-paille** [comment]
@ashwinvis way to go! You have several failing test cases on travis, can you handle that? Also, you'll need to squash your commits once they validate ;-).

File: pythran/config.py (hunk 4 above)
Comment (ashwinvis) [reply pythran/config.py]: It is what I intended, a sort of enhancement for the `pythran-config -v` behaves like a shortcut intead of `pythran-config --compiler --cflags --libs`. And `pythran-config -vv` makes the same output more readable as follows:

```sh
❯❯❯ pythran-config --no-python -vv > script.sh
❯❯❯ cat script.sh                                                  (pythran) enh/pythranrc-sanity-check ◼
     CXX = clang++ 
  CFLAGS = -D__PYTHRAN__=3 -DPYTHRAN_BLAS_OPENBLAS -I/home/avmo/src/pythran/pythran -I/home/avmo/src/pythran/pythran/pythonic/patch 

```

Which is more readable and can be output to a file to be easily modified to behave like a bash script if needed. 

If you don't like it I can revert it.

**Comment by ashwinvis** [comment]
> @ashwinvis way to go! You have several failing test cases on travis, can you handle that? Also, you'll need to squash your commits once they validate ;-).

I have squashed it, but the test cases are beyond my understanding. I don't understand what happened here:

https://travis-ci.com/serge-sans-paille/pythran/jobs/163867306
https://travis-ci.com/serge-sans-paille/pythran/jobs/163867308
https://travis-ci.com/serge-sans-paille/pythran/jobs/163867310

**Comment by serge-sans-paille** [comment]
> If you don't like it I can revert it.

The problem is that `pythran-config` is meant to be usable that way:

g++ `pythran-config --cflags --libs` stuff.cpp

and you're breaking that :-/

**Comment by ashwinvis** [comment]
I see your point, will change those into `logger.info` calls then.

File: pythran/config.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ def init_cfg(sys_file, platform_file, user_file):
 
     cfgp = ConfigParser()
     for required in (sys_config_path, platform_config_path):
-        cfgp.readfp(open(required))
+        cfgp.read([required])
     cfgp.read([user_config_path])
```
Comment (ashwinvis) [inline pythran/config.py]: I hope there was a good reason why this was not written as:

```python
cfgp.read((sys_config_path, platform_config_path))
cfgp.read([user_config_path])
# OR
cfgp.read((sys_config_path, platform_config_path, user_config_path))
```

Comment (ashwinvis) [reply pythran/config.py]: The change can be traced back to 6a310b5

**Comment by ashwinvis** [comment]
@serge-sans-paille All the failed tests are related to openmp, and it beats me what this PR has to do with it. Will you be able to take it from here?

**Comment by serge-sans-paille** [comment]
@ashwinvis yep, I'll do that

**Comment by serge-sans-paille** [comment]
Obsoleted by #1145
